pkg/clusteragent/admission/controllers/webhook/common.go

```diff
@@ -16,26 +16,14 @@ import (
 
 // buildLabelSelectors returns the mutating webhooks object selector based on the configuration
 func buildLabelSelectors(useNamespaceSelector bool) (namespaceSelector, objectSelector *metav1.LabelSelector) {
-	var labelSelector metav1.LabelSelector
-
-	if config.Datadog.GetBool("admission_controller.mutate_unlabelled") {
-		// Accept all, ignore pods if they're explicitly filtered-out
-		labelSelector = metav1.LabelSelector{
-			MatchExpressions: []metav1.LabelSelectorRequirement{
-				{
-					Key:      common.EnabledLabelKey,
-					Operator: metav1.LabelSelectorOpNotIn,
-					Values:   []string{"false"},
-				},
-			},
-		}
-	} else {
-		// Ignore all, accept pods if they're explicitly allowed
-		labelSelector = metav1.LabelSelector{
-			MatchLabels: map[string]string{
-				common.EnabledLabelKey: "true",
+	labelSelector := metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				Key:      common.EnabledLabelKey,
+				Operator: metav1.LabelSelectorOpNotIn,
+				Values:   []string{"false"},
 			},
-		}
+		},
 	}
```

Comment:
I think we should keep the current `admission_controller.mutate_unlabelled` option and the `else`.
then add a new option (disable) by default. This new settings can take precedence on the other solution

something like
```golang
     if config.Datadog.GetBool("admission_controller.mutate_unlabelled") {
       labelSelector = metav1.LabelSelector{
		MatchExpressions: []metav1.LabelSelectorRequirement{
			{
				Key:      common.EnabledLabelKey,
				Operator: metav1.LabelSelectorOpNotIn,
				Values:   []string{"false"},
			},
		},
      }
     }
     else if config.Datadog.GetBool("admission_controller.mutate_unlabelled") {
		// Accept all, ignore pods if they're explicitly filtered-out
		labelSelector = metav1.LabelSelector{
			MatchExpressions: []metav1.LabelSelectorRequirement{
				{
					Key:      common.EnabledLabelKey,
					Operator: metav1.LabelSelectorOpNotIn,
					Values:   []string{"false"},
				},
			},
		}
	} else {
		// Ignore all, accept pods if they're explicitly allowed
		labelSelector = metav1.LabelSelector{
			MatchLabels: map[string]string{
				common.EnabledLabelKey: "true",
			},
		}
	}
````

Reply:
Latest commit implements this logic